File: 1593 B. Make it Divisible by 25/Solution.cpp
// 325061256
#include<bits/stdc++.h>
using namespace std;

void solve(){
    long long n; cin>>n;
    string s = to_string(n);
    vector<string> div = {"00","25","50","75"};
    int ans = s.size();
    for(string str : div){
        int i = s.size()-1;
        while(i>=0 && s[i]!=str[1]){
            i--;
        }
        if(i<0){continue;}
        i--;
        while(i>=0 && s[i]!=str[0]){
            i--;
        }
        if(i<0){continue;}

        int rem = (s.size()-1) - i - 1;
        ans = min(ans,rem);
    }
    cout<<ans<<endl;
}

int main(){
    int t; cin>>t;
    while(t--){
        solve();
    }
    return 0;
}

File: 1593 B. Make it Divisible by 25/Solution.py
# 325065520
def solve():
    n = int(input())
    s = str(n)
    ans = len(s)
    div = ["00","25","50","75"]
    for st in div:
        i = len(s)-1
        while i>=0 and s[i]!=st[1]:
            i -= 1
        if i<0:
            continue
        i -= 1
        while i>=0 and s[i]!=st[0]:
            i -= 1
        if i<0:
            continue
        rem = len(s)-1-i-1
        ans = min(ans,rem)
    print(ans)

def main():
    t = int(input())
    for _ in range(t):
        solve()

if __name__ == "__main__":
    main()
